### PR TITLE
CPM-854: Add Enabled condition

### PR DESCRIPTION
**Pull request by pierallard**

### CODE REVIEW

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx [hunk 1]
```diff
@@ -1,46 +1,75 @@
...
+    conditions.map(condition => ({
+      id: uuid(),
+      ...condition,
+    }))
+  );
+
+  const removeIds: (conditionsWithId: ConditionsWithIdentifier) => Conditions = conditionsWithId => {
+    return conditionsWithId.map(conditionWithId => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const {id, ...condition} = conditionWithId;
```
Comment (antoinearrive) [inline components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx:31]: why get an ID and not use it?

File: components/identifier-generator/front/src/feature/models/conditions/conditions.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import {Enabled} from './enabled';
+
+type Condition = {
+  type: CONDITION_NAMES;
+} & Enabled;
```
Comment (antoinearrive) [inline components/identifier-generator/front/src/feature/models/conditions/conditions.ts:5]: just to know, I don't know this syntax {}&, does it add the enabled object if it is not null?

**Review by antoinearrive — APPROVED** [review]

File: components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx (hunk 1 above)
Comment (pierallard) [reply components/identifier-generator/front/src/feature/tabs/SelectionTab.tsx:31]: The notation 
```const {id, ...condition} = conditionWithId```
will "extract" the `id` property from `conditionWithId`.
This is the "beautiful way" to remove a key from an object in JS.

File: components/identifier-generator/front/src/feature/models/conditions/conditions.ts (hunk 2 above)
Comment (pierallard) [reply components/identifier-generator/front/src/feature/models/conditions/conditions.ts:5]: These are not objects, these are types.
If I declare `type AB = { a: string, b: string }` and `type BC = { a: string, b: string}`, so `type ABC = AB & AC` will declare a ` {a: string, b: string, c: string }`.
Here, I'm saying "A `Condition` has a `type` property (in the list of allowed names), and can be an `Enabled` condition.
In the future, we will have something like this
```
type Condition = {
  type: CONDITION_NAMES;
} & (Enabled | FamilyCondition | SimpleSelectCondition | AnotherCondition);
```

Comment (antoinearrive) [reply components/identifier-generator/front/src/feature/models/conditions/conditions.ts:5]: ok thanks for the explanation

**Review by Beninho — APPROVED** [review]

**Review by mmetayer — APPROVED** [review]

**Review by cemorin — APPROVED** [review]

**Pull request merged**